### ISSUE TITLE
IndexOptimize performance improvement required for databases with high number of indexes

### ISSUE THREAD

**Issue by jrueschenschmidt** [olahallengren/sql-server-maintenance-solution#304]
IndexOptimize performance improvement required for database with high number of indexes. Currently, if a database contains a very high number of indexes (e.g. > 250.000), initial step where stored procedure collects data what needs to be processed takes hours, even if @Indexes parameter is set to narrow down the work.

I started a discussion in April 2019 on https://dba.stackexchange.com/questions/234520/sql-server-index-and-statistics-maintenance-performance-question already. A member suggested a modification of IndexOptimize stored procedure. I implemented his suggestion in April and it worked well. Would be nice to have it implemented into regular IndexOptimize stored procedure. It would be a big improvement for all DBA’s dealing with databases where high number of indexes exist.

**Comment by ShadowXVII** [comment]
Yep, we've had to move away from these scripts. They don't scale.

**Comment by olahallengren** [comment]
I have recently made multiple performance improvements to `IndexOptimize`, specifically for databases with a very large number of indexes and statistics.
